src/main/java/org/opensearch/jobscheduler/rest/RestGetJobIndexAction.java

```diff
@@ -0,0 +1,135 @@
+
+import org.opensearch.jobscheduler.utils.JobDetailsService;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.BytesRestResponse;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/**
+ * This class consists of the REST handler to GET job index from extensions.
+ */
+public class RestGetJobIndexAction extends BaseRestHandler {
+
+    public static final String GET_JOB_INDEX_ACTION = "get_job_index_action";
+
+    private final Logger logger = LogManager.getLogger(RestGetJobIndexAction.class);
+
+    public JobDetailsService jobDetailsService;
+
+    public RestGetJobIndexAction(final JobDetailsService jobDetailsService) {
+        this.jobDetailsService = jobDetailsService;
+    }
+
+    @Override
+    public String getName() {
+        return GET_JOB_INDEX_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return unmodifiableList(
+            asList(new Route(PUT, String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_get/_job_index")))
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        XContentParser parser = restRequest.contentParser();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+
+        GetJobIndexRequest getJobIndexRequest = GetJobIndexRequest.parse(parser);
+
+        final JobDetails[] jobDetailsResponseHolder = new JobDetails[1];
+
+        String jobIndex = getJobIndexRequest.getJobIndex();
+        String jobParameterAction = getJobIndexRequest.getJobParameterAction();
+        String jobRunnerAction = getJobIndexRequest.getJobRunnerAction();
+        String extensionId = getJobIndexRequest.getExtensionId();
+
+        CompletableFuture<JobDetails[]> inProgressFuture = new CompletableFuture<>();
+
+        jobDetailsService.processJobDetailsForExtensionId(
+            jobIndex,
+            null,
+            jobParameterAction,
+            jobRunnerAction,
+            extensionId,
+            JobDetailsService.JobDetailsRequestType.JOB_INDEX,
+            new ActionListener<>() {
+                @Override
+                public void onResponse(JobDetails jobDetails) {
+                    jobDetailsResponseHolder[0] = jobDetails;
+                    inProgressFuture.complete(jobDetailsResponseHolder);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    logger.info("could not process job index", e);
+                    jobDetailsResponseHolder[0] = null;
```

Comment:
This isn't necessary as the value is already null from the array instantiation.  It's reasonable from a readability standpoint to add a comment here indicating this remains null.

Reply:
done